bionic/persistence.py

```diff
@@ -754,43 +750,35 @@ class LocalFilesystem:
     to local disk.
     """
 
-    def __init__(self, inventory_dir, tmp_dir):
+    def __init__(self, inventory_dir):
         self.root_url = url_from_path(inventory_dir)
-        self.tmp_root_path = tmp_dir
+        self._fs = fsspec.filesystem("file")
 
     def exists(self, url):
-        return path_from_url(url).exists()
+        return self._fs.exists(unquote(url))
 
     def search(self, url_prefix):
-        path_prefix = path_from_url(url_prefix)
-        if not path_prefix.is_dir():
-            return []
-
-        return [
-            url_from_path(path_prefix / sub_path)
-            for sub_path in path_prefix.glob("**/*")
-        ]
+        # This endpoint is a glob **/* search.
+        glob_url = unquote(url_prefix) + "**/*"
+        return ["file://" + url for url in self._fs.glob(glob_url)]
 
     def rm(self, url):
-        path = path_from_url(url)
-        path.unlink()
+        self._fs.rm(unquote(url))
 
     def write_bytes(self, content_bytes, url):
-        path = path_from_url(url)
-        ensure_parent_dir_exists(path)
-        ensure_dir_exists(self.tmp_root_path)
-        working_dir = Path(tempfile.mkdtemp(dir=str(self.tmp_root_path)))
-        try:
-            working_path = working_dir / "tmp_file"
-            working_path.write_bytes(content_bytes)
-
-            working_path.rename(path)
+        ensure_parent_dir_exists(path_from_url(url))
 
-        finally:
-            shutil.rmtree(str(working_dir))
+        with self._fs.transaction:
```

Comment:
Wondering, why is it necessary to use a `transaction` here?

Reply:
We write the file to a temporary location and then move it to the cache location to avoid data corruption in case the write somehow failed or stopped. We can now do the same thing using `transaction` instead.